src/Common/ItechArt.Common/Validation/Abstractions/IValidator.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace ItechArt.Common.Validation.Abstractions;
+
+public interface IValidator<TEntity, TError>
```

Comment:
let's add base implementation which:
1. will have abstract `TError? Validate(TEntity entity)`
2. will implement `ValidatationResult<TError> Validate(TEntity entity)` by invoking method from p.1 and converting nullable error to ValidationResult